pkg/interface/chat/src/js/store.js

```diff
@@ -9,7 +9,20 @@ import { LocalReducer } from '/reducers/local.js';
 
 class Store {
   constructor() {
-    this.state = {
+    this.state = this.initialState();
+
+    this.initialReducer = new InitialReducer();
+    this.permissionUpdateReducer = new PermissionUpdateReducer();
+    this.contactUpdateReducer = new ContactUpdateReducer();
+    this.chatUpdateReducer = new ChatUpdateReducer();
+    this.inviteUpdateReducer = new InviteUpdateReducer();
+    this.metadataReducer = new MetadataReducer();
+    this.localReducer = new LocalReducer();
+    this.setState = () => {};
+  }
+
+  initialState() {
+    return {
```

Comment:
Any reason for programming it to ignore future “initial” objects instead of reducing them to diffs? Or even clobbering the props of the same type? We might drop updates this way, no?

Reply:
I'm not sure what you mean. `Subscription` sends `Store` a clear event that resets `Store` entirely to its initial state. Then `Subscription` re-establishes all subscriptions. It's the equivalent of a page reload.

Reply:
Oh! Ok. I thought we were just dropping future initials based on the glance here. Ok cool